File: MatricesJava/src/Ejercicio4.java
package paquete;

import java.util.Scanner;

public class Ejercicio4 {
    static public void main(String[] args){
        Scanner scanner = new Scanner(System.in);
        int cantidadFilas, cantidadColumnas, columnaMayor = 0; 
        System.out.print("Ingrese la cantidad de filas: "); cantidadFilas = scanner.nextInt();
        System.out.print("Ingrese la cantidad de columnas: "); cantidadColumnas = scanner.nextInt();
        float matriz[][] = new float[cantidadFilas][cantidadColumnas]; float sumaColumna = 0, columnaConSumaMayor = 0, primeraSuma = 0;
        
        for (int i = 0; i < cantidadFilas; i++) {
            for (int j = 0; j < cantidadColumnas; j++) {
                System.out.print("Ingrese un numero para la posicion: (" +
                        (i+1) + "-" + (j+1) + "): "); matriz[i][j] = scanner.nextInt();
            }
        }
        for (int i = 0; i < cantidadColumnas; i++) {
            sumaColumna = 0;
            for (int j = 0; j < cantidadFilas; j++) {
                sumaColumna += matriz[j][i];
            }
            primeraSuma = 0;
            if(sumaColumna >= columnaConSumaMayor){
                columnaConSumaMayor = sumaColumna;
                columnaMayor = i;
            }
        }
        System.out.println("La columna mayor es: " + columnaMayor);
        System.out.println("Suma Columna: " + columnaConSumaMayor);
    }
}

File: MatricesC++/Ejercicio4.cpp
#include<iostream>
#include<conio.h>

using namespace std;

int main(){
	/*Hacer un programa que llene una matriz de 20*20. Sumar las columnas e imprimir que columna tuvo la maxima suma y la suma de esa columna*/
	int matriz[10][10];
	int cantidadFilas  = 0, cantidadColumnas = 0,  sumaMayorColumnas = 0;
	int sumaColumna = 0, posicionColumnaMayorSuma = 0; 
	cout<<"Por favor ingrese la cantiadd de filas: "; cin >> cantidadFilas; 
	cout<<"Por favor ingrese la cantidad de columnas: "; cin >> cantidadColumnas;
	for(int i = 0; i < cantidadFilas; i++){
		for(int j = 0; j < cantidadColumnas; j++){
			cout<<"Por favor ingrese un numero para la posicion: ("<<i+1<<"-"<<j+1<<"): "; cin >> matriz[i][j];
		}
	}	
	for(int i = 0; i < cantidadColumnas; i++){
		sumaColumna = 0;
		for(int j = 0; j < cantidadFilas; j++){
			sumaColumna += matriz[j][i];
		}
		if(sumaColumna > sumaMayorColumnas){
			sumaMayorColumnas = sumaColumna; 
			posicionColumnaMayorSuma = i;
		}
	}

	cout<<"La posicion de la columna con mayor suma es: "<<posicionColumnaMayorSuma+1;
	cout<<endl<<"La suma de su columna es: "<<sumaMayorColumnas;
	getch();
	return 0; 
}
